refactor(router): remove duplicate type from the LoadChildrenCallback

rafaelss95 opened the pull request:
This removes the duplicated `Type<any>` from the `LoadChildrenCallback` type literal used in Route#loadChildren.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] Other... Please describe:


## What is the current behavior?
The _type_ `LoadChildrenCallback` has a duplicate `Type<any>`.


## What is the new behavior?
The _type_ `LoadChildrenCallback` has no duplicate `Type<any>`.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

rafaelss95 commented on packages/router/src/config.ts:
```diff
@@ -110,8 +110,8 @@ export type ResolveData = {
-export type LoadChildrenCallback = () => Type<any>|NgModuleFactory<any>|Observable<Type<any>>|
-    Promise<NgModuleFactory<any>|Type<any>|any>;
+export type LoadChildrenCallback = () =>
+    Type<any>|NgModuleFactory<any>|Observable<Type<any>>|Promise<NgModuleFactory<any>|any>;
```
I just noticed that it has `| any`, which makes all other types redundant. Is it expected?

JoostK commented:
The second occurence of `Type` appears in the generic type argument to `Promise`, so it doesn't seem duplicate to me.

I don't know about the `any` but it's probably like that for historic reasons.

rafaelss95 committed:
refactor(router): remove duplicate type from the LoadChildrenCallback

This removes the duplicated `Type<any>` from the `LoadChildrenCallback` type literal used in Route#loadChildren

rafaelss95 commented:
> The second occurence of `Type` appears in the generic type argument to `Promise`, so it doesn't seem duplicate to me.
> 
> I don't know about the `any` but it's probably like that for historic reasons.

Aha, you are correct. I quickly looked through the documents and made the changes.
